terracotta/drivers/terracotta_driver.py

```diff
@@ -0,0 +1,184 @@
+        self.meta_store.create(keys=keys, key_descriptions=key_descriptions)
+
+    def connect(self, verify: bool = True) -> contextlib.AbstractContextManager:
+        return self.meta_store.connect(verify=verify)
+
+    @requires_connection
+    def get_keys(self) -> OrderedDict:
+        return self.meta_store.get_keys()
+
+    @requires_connection
+    def get_datasets(self, where: MultiValueKeysType = None,
+                     page: int = 0, limit: int = None) -> Dict[Tuple[str, ...], Any]:
+        return self.meta_store.get_datasets(
+            where=self._standardize_multi_value_keys(where, requires_all_keys=False),
+            page=page,
+            limit=limit
+        )
+
+    @requires_connection
+    def get_metadata(self, keys: ExtendedKeysType) -> Dict[str, Any]:
+        keys = self._standardize_keys(keys)
+
+        metadata = self.meta_store.get_metadata(keys)
+
+        if metadata is None:
+            # metadata is not computed yet, trigger lazy loading
+            dataset = self.get_datasets(keys)
+            if not dataset:
+                raise exceptions.DatasetNotFoundError('No dataset found')
+
+            path = squeeze(dataset.values())
+            metadata = self.compute_metadata(path, max_shape=self.LAZY_LOADING_MAX_SHAPE)
+            self.insert(keys, path, metadata=metadata)
+
+            # ensure standardized/consistent output (types and floating point precision)
+            metadata = self.meta_store.get_metadata(keys)
+            assert metadata is not None
+
+        return metadata
+
+    @requires_connection
+    def insert(
+        self, keys: ExtendedKeysType,
+        path: Any, *,
+        override_path: str = None,
+        metadata: Mapping[str, Any] = None,
+        skip_metadata: bool = False,
+        **kwargs: Any
+    ) -> None:
+        keys = self._standardize_keys(keys)
+
+        if metadata is None and not skip_metadata:
+            metadata = self.compute_metadata(path)
+
+        self.meta_store.insert(
+            keys=keys,
+            path=override_path or path,
+            metadata=metadata,
+            **kwargs
+        )
+
+    @requires_connection
+    def delete(self, keys: ExtendedKeysType) -> None:
+        keys = self._standardize_keys(keys)
+
+        self.meta_store.delete(keys)
+
+    def get_raster_tile(self, keys: ExtendedKeysType, *,
+                        tile_bounds: Sequence[float] = None,
+                        tile_size: Sequence[int] = (256, 256),
+                        preserve_values: bool = False,
+                        asynchronous: bool = False) -> Any:
+        path = squeeze(self.get_datasets(keys).values())
+
+        return self.raster_store.get_raster_tile(
+            path=path,
+            tile_bounds=tile_bounds,
+            tile_size=tile_size,
+            preserve_values=preserve_values,
+            asynchronous=asynchronous,
+        )
+
+    def compute_metadata(self, path: str, *,
```

Comment:
Any reason why `compute_metadata` takes a path to identify the dataset, while `get_raster_tile` takes a sequence of keys? I would think they should both take keys, for consistency?

Reply:
Nevermind, I guess that `TerracottaDriver.get_metadata()` is the proper interface here.

But then I don't think there is any reason to expose `compute_metadata()` on `TerracottaDriver`?

Reply:
It has to be exposed for this recipe:

```python
mtd = driver.compute_metadata(raster_file, extra_metadata={"custom": "foo"})
driver.insert(raster_file, metadata=mtd)
```

`compute_metadata` is used to compute *before* insertion, so keys are meaningless at this stage.

Reply:
Makes sense. I also did make sure to add documentation for this method.